### ISSUE TITLE
Arduino OPTA port

### ISSUE THREAD

**Issue by patkepa** [budulinek/arduino-modbus-rtu-tcp-gateway#32]
Hello!
I'm interested in porting this project to the Arduino OPTA(STM32H747XI)
Are there any suggestions  in regards to replacing the Atmega based code of this project, for STM32?

Thanks!

**Comment by budulinek** [comment]
What ethernet chip is used in Arduino OPTA? Which ethernet chip do you want to use with STM32?

The code is tailor-made for the Wiznet W5100/W5200/W5500 ethernet chip. It would be very difficult to "port" the code to different ethernet chip.

**Comment by papercodeIN** [comment]
Hi Creator, 

Could you guide me how to implement this in wiznet w5100s-evb-pico (Raspberry Pi Pico) ?

**Comment by budulinek** [comment]
> Hi Creator,
> 
> Could you guide me how to implement this in wiznet w5100s-evb-pico (Raspberry Pi Pico) ?

I have no idea on how to port it to RPi. Some functions and libraries used by this project are specific to W5X00 ethernet chips (Ethernet.h library), some functions I use are specific to ATmega chips (random number generator, reset function, also probably the Serial library...).

Disclosure: I am a 100% hobbyist coder with no formal coding education. I made this gateway for fun and for my self-education. A kind of challenge - what else can I fit into a very limited flash and RAM resources of the ATmega chip. I did not and do not have an intention to port this project to bigger and more powerful platforms.
